Sources/NIO/Bootstrap.swift

```diff
@@ -219,7 +219,16 @@ public final class ServerBootstrap {
     ///
     /// - parameters:
     ///     - unixDomainSocketPath: The _Unix domain socket_ path to bind to. `unixDomainSocketPath` must not exist, it will be created by the system.
-    public func bind(unixDomainSocketPath: String) -> EventLoopFuture<Channel> {
+    ///     - cleanupExistingSocketFile: Whether to cleanup an existing socket file at `path`.
+    public func bind(unixDomainSocketPath: String, cleanupExistingSocketFile: Bool = true) -> EventLoopFuture<Channel> {
```

Comment:
We cannot add new parameters to existing Swift functions: that's a breaking change. The reason is that functions can be referred to by fully-qualified name, e.g.

```swift
let someFunc = ServerBootstrap.bind(unixDomainSocketPath:)
```

With this change the above function no longer exists, and the above code would break. We need a _new_ function instead.

Reply:
fair, should we perhaps use `ChannelOption` instead then? Or even just piggyback on `SO_REUSEADDR`?

Reply:
I'm disinclined to piggyback on `so_reuseaddr` tbh, it isn't the semantic we want. We could use a channel option but I think for now it's better to just add the new functions to do this. If we find a second kind of socket that has similar semantics we can consider merging them together into a channel option.